database/src/models/account.rs

```diff
@@ -221,4 +224,33 @@ impl Account {
             .execute(&**db_conn)
             .and(Ok(()))
     }
+
+    pub fn display_name_or_username<'a>(&'a self) -> &'a String {
+        &self.username
```

Comment:
This is not currently correct, ensure that display name is returned as well.

Reply:
you likely want something like
```rust
self.display_name.as_ref().unwrap_or(&self.username)
```

also, you don't need the `'a` lifetime bound in that function - lifetime elision will replace it. moreover, you should be returning an `&str` ([see TRPL](https://doc.rust-lang.org/book/second-edition/ch08-02-strings.html)).

Reply:
Done, thanks - I'm embarrassed to say I had some very wrong assumptions about what `String` vs. `str` was, now corrected.